modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcTransactionTest.java

```diff
@@ -0,0 +1,307 @@
+        }
+    }
+
+    /**
+     * Tests {@link Connection#commit()} method behaviour.
+     *
+     * <p>Calling {@code commit} is expected to throw an exception if called in auto-commit mode or on a closed connection,
+     *
+     * @throws SQLException If failed.
+     */
+    @Test
+    public void testCommit() throws Exception {
+        try (Connection conn = DriverManager.getConnection(URL)) {
+            SQLException ex = assertThrows(SQLException.class, conn::commit);
+            MatcherAssert.assertThat(ex.getMessage(), containsString("Transaction cannot be committed explicitly in auto-commit mode."));
+
+            conn.setAutoCommit(false);
+            conn.commit();
+
+            conn.close();
+
+            // Exception when called on closed connection
+            checkConnectionClosed(conn::commit);
+        }
+    }
+
+    /**
+     * Tests {@link Connection#rollback()} method behaviour.
+     *
+     * <p>Calling {@code rollback} is expected to throw an exception if called in auto-commit mode or on a closed connection,
+     *
+     * @throws SQLException If failed.
+     */
+    @Test
+    public void testRollback() throws Exception {
+        try (Connection conn = DriverManager.getConnection(URL)) {
+            // Should not be called in auto-commit mode
+            SQLException rollbackEx = assertThrows(SQLException.class, conn::rollback);
+            MatcherAssert.assertThat(rollbackEx.getMessage(),
+                    containsString("Transaction cannot be rolled back explicitly in auto-commit mode."));
+
+            conn.setAutoCommit(false);
+            conn.rollback();
+
+            conn.close();
+
+            // Exception when called on closed connection
+            checkConnectionClosed(conn::rollback);
+        }
+    }
+
+    /**
+     * Ensures that {@link Statement#executeUpdate(String)} supports explicit transaction.
+     *
+     * <p>It is expected that in non auto-commit mode, data updates can be rolled back and
+     * are invisible outside of an active transaction until they are committed.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testExecuteUpdate() throws Exception {
+        try (Statement stmt = conn.createStatement()) {
+            checkUpdate(2, 1, (id, ignore) -> stmt.executeUpdate(String.format(SQL_INSERT, id, "name-" + id)));
+        }
+    }
+
+    /**
+     * Ensures that {@link PreparedStatement#executeUpdate()} supports explicit transaction.
+     *
+     * <p>It is expected that in non auto-commit mode, data updates can be rolled back and
+     * are invisible outside of an active transaction until they are committed.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testExecuteUpdatePrepared() throws Exception {
+        try (PreparedStatement pstmt = conn.prepareStatement(SQL_INSERT_PREPARED)) {
+            checkUpdate(2, 1, (id, ignore) -> {
```

Comment:
what about tests for autoCommit=true mode?

Reply:
It seems, that currently we have enough tests for auto-commit mode - ItJdbcStatementSelfTest, ItJdbcBatchSelfTest, and so on.

Do you suggest adding some checks here for this mode as well?

Reply:
ok, fair enough